src/universe/index.ts

```diff
@@ -254,6 +255,81 @@ export interface UniverseAnalyticsEventsOptions {
   end: string
   datepart?: string
 }
+export interface UniverseMe {
+  (skipInterceptors?: boolean): Promise<MeData | never>
+  addTemplateFavorite: (favorite: StaffMessageTemplateFavorite) => Promise<staff.StaffRawPayload | never>
+  removeTemplateFavorite: (id: string) => Promise<staff.StaffRawPayload | never>
+}
+
+const me: UniverseMe = async function (this: Universe, skipInterceptors = false): Promise<MeData | never> {
+  try {
+    const opts = {
+      method: 'GET',
+      url: `${this.universeBase}/api/v0/me`,
+      skipInterceptors
+    }
+
+    const response = await this.http.getClient()(opts)
+
+    this.setCachedMeData(response?.data?.data)
+
+    return response?.data?.data
+  } catch (error) {
+    throwExceptionFromCommonError(error as ClientError)
+
+    throw new UniverseMeError(undefined, { error })
+  }
+}
+
+me.addTemplateFavorite = async function (this: Universe, favorite: StaffMessageTemplateFavorite): Promise<staff.StaffRawPayload | never> {
+  const me = await this.me()
+
+  if (!me.staff.id) throw new UniverseForbiddenError()
+
+  const opts: AxiosRequestConfig = {
+    method: 'PATCH',
+    url: `${this.universeBase}/api/v0/staff/${me.staff.id}`,
+    data: []
+  }
+
+  if (!Array.isArray(me.preferences?.message_template_favorites)) {
+    opts.data.push({
+      op: 'add',
+      path: '/preferences/message_template_favorites',
+      value: []
+    })
+  } else if ((me.preferences?.message_template_favorites as staff.StaffMessageTemplateFavorite[]).length >= 7) {
+    throw new UniverseBadRequestError('Cannot have more than 7 message template favorites.', { response: { status: 400 } })
+  }
+
+  opts.data.push({
+    op: 'add',
+    path: '/preferences/message_template_favorites/-',
+    value: favorite
+  })
+
+  return await this.patchStaffPreferences(opts)
+}
+
+me.removeTemplateFavorite = async function (this: Universe, id: string): Promise<staff.StaffRawPayload | never> {
+  const me = await this.me()
+
+  if (!me.staff.id) throw new UniverseForbiddenError()
+
+  const opts: AxiosRequestConfig = {
+    method: 'PATCH',
+    url: `${this.universeBase}/api/v0/staff/${me.staff.id}`,
+    data: []
+  }
+
+  opts.data.push({
+    op: 'replace',
+    path: '/preferences/message_template_favorites',
+    value: me.preferences?.message_template_favorites?.filter(f => f.id !== id)
```

Comment:
also we prbly want to test those `removeTemplateFavorite` and `addTemplateFavorite` methods... Not sure how the PATCH works with replace and value being undefined. It might reaplace it really, but it could actually ignore the value completely...